root/etc/systemd/system/nm-connected.service

```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=Network Manager Wait Connected
```

Comment:
Maybe something like this could be better: `Wait for NetworkManager DHCP activation` because without `-s` this is really awaiting DHCP activation rather than awaiting connection.